#37 : Added variables for Debian and Ubuntu

Zauxst opened the pull request:
- Moved the dependencies variables from `default` folder to `vars` folder, this would help in the future to support multiple platforms that are outside of Debian ecosystem.
- Removed `gnupg2` from Ubuntu install.
- Added to `README.md` the requirement section which specifies for Ubuntu18.04 users to enable Universe.

**This would only fix "part" of the problem**, _maybe a further separation of tasks and variables needs to be done to ensure full compatibility with Ubuntu18.04 clean server installs._

Initially I wanted to add the `universe` repo sources list but I realized that it was out of scope of your ethic for this role; it could've impacted people that work on different sources and it is, in the end, out of the scope of this role.

I know we have discussed in the topic issue regarding `gnupg2` but it is not required in a base install as opposed to `python-pip`s importance.
Python-pip is as well in the `universe` repo source list, so the warning should be on the `README.md` section until Canonical offers a way to install the repos during a clean install of Ubuntu18.04.

nickjj commented:
Thanks for the PR but that means users won't be able to configure which packages get installed, which I like having as a role default as Docker's dependency requirements tend to change sometimes between versions.

This is why I mentioned in the other issue that it complicates both the configuration and implementation, as it would need to expose multiple new config options -- and ideally you'd want them all to be rolled up into a single list at the task level so they can be handled by 1x apt module call (the common and distro specific packages).

`python-pip` is a necessary evil because Ansible's `docker_login` module requires pip installing docker btw.

Zauxst commented:
In light of your decision i've modified a bit the variable structure:
- added back the "user defined" variable for dependencies and initialized it as empty array to keep the same structure as before
- In the `vars` folder renamed the variables with a underscore (`_`) in front.

nickjj commented:
Closing this due to there being a lot more that was added / changed in v1.6.0. It also addressed this PR's issue but in a different way.
